src/group/mod.rs

```diff
@@ -131,16 +127,16 @@ impl<'a> Groups {
         })
     }
 
-    /// Unset the delegate by delegate id.
+    /// Unset the delegate by delegate id. If the peer is not delegate, do nothing.
     pub(crate) fn remove_delegate(&mut self, delegate: u64) {
-        if self.indexes.remove(&delegate).is_some() {
+        if self.bcast_targets.remove(&delegate).is_some() {
+            self.indexes.remove(&delegate);
```

Comment:
I think just set the `delegate` to `INVALID_ID` is enough?

Reply:
When next time we call `pick_delegate`, this could become delegate again, It's unexpected.

Reply:
```suggestion
            // Remove the delegate from the group system since it's temporary unreachable from the leader's view.
            // And the peer will be re-added after receiving a message from it.
            self.indexes.remove(&delegate);
```